ols/app/models/config.py

```diff
@@ -82,8 +84,22 @@ def __init__(self, data: Optional[dict] = None) -> None:
         if data is None:
             return
         self.name = data.get("name", None)
+        # Default provider type to be the provider name, unless
+        # specified explicitly.
+        self.type = data.get("type", self.name)
+        if self.type not in constants.SUPPORTED_PROVIDER_TYPES:
```

Comment:
str(self.type).lower() , as in llm_loader.py? or be UNIXy and don't lower() in llm_loader.py?

Reply:
actually, with addition of this code ^^^, the lower() in llm_loader.py doesn't do anything, only lowercase self.type will pass validation.

Reply:
probably needs a follow up, we should be case-insensitive on matching the provider type.